extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaStringHeaderReader.java

```diff
@@ -48,10 +48,17 @@ public KafkaStringHeaderReader(Headers headers,
   public List<Pair<String, Object>> read()
   {
     List<Pair<String, Object>> events = new ArrayList<>();
+
     for (Header hdr : headers) {
-      String s = new String(hdr.value(), this.encoding);
-      String newKey = this.headerLabelPrefix + hdr.key();
-      events.add(Pair.of(newKey, s));
+      byte[] value = hdr.value();
+
+      if (value != null) {
+        String s = new String(value, this.encoding);
+        String newKey = this.headerLabelPrefix + hdr.key();
+        events.add(Pair.of(newKey, s));
+      } else {
+        log.warn("Header '" + hdr.key() + "' skipped because it had no value");
+      }
```

Comment:
As suggested in this comment, https://github.com/apache/druid/issues/16177#issuecomment-2009731844, we should just add a null value rather than skipping this header altogether.

```suggestion
      String s = value == null ? null : new String(value, this.encoding);
      String newKey = this.headerLabelPrefix + hdr.key();
      events.add(Pair.of(newKey, s));
```